File: DesignPatternsDotNet/CommandPattern/Commands/TurnMobileOn.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CommandPattern.Commands
{
  public class TurnMobileOn : ICommand{
	private IReceiver myReceiver  = null;
	public TurnMobileOn(IReceiver aReceiver ){
		myReceiver= aReceiver ;		
	}
	public void Execute() {
		myReceiver.On();	
	}
	public void Undo() {
		myReceiver.Off();
	}
}
}

File: DesignPatternsJava/src/com/command/TurnMobileOn.java
package com.command;

public class TurnMobileOn implements ICommand{
	private IReceiver myReceiver  = null;
	public TurnMobileOn(IReceiver aReceiver ){
		myReceiver= aReceiver ;		
	}
	public void execute() {
		myReceiver.on();	
	}
	public void undo() {
		myReceiver.off();
	}
}